### PR TITLE
fix: fetch balances when opening token selector in LP page

### PR DESCRIPTION
**Pull request by just-toby**
## Description

we're not properly fetching the user's token balances on the LP page. this should happen when hovering over the token input similar to the swap component.

to fix we just want to wrap the token input with a `PrefetchBalancesWrapper` which triggers the fetch

_Slack thread:_ https://uniswapteam.slack.com/archives/C04D07TQBT4/p1694811755326329

### Reproducing the error

1. go to add liquidity, v2 or v3 - notice that the token balances don't load when you open the selector

### QA (ie manual testing)

- [x] repeat step 1 above but notice that the balances load now

### CODE REVIEW

File: src/components/CurrencyInputPanel/index.tsx [hunk 1]
```diff
@@ -230,45 +231,50 @@ export default function CurrencyInputPanel({
...
-                            '...' +
-                            currency.symbol.slice(currency.symbol.length - 5, currency.symbol.length)
-                          : currency?.symbol) || <Trans>Select a token</Trans>}
-                      </StyledTokenName>
-                    )}
-                  </RowFixed>
-                  {onCurrencySelect && <StyledDropDown selected={!!currency} />}
-                </Aligner>
-              </CurrencySelect>
+              <PrefetchBalancesWrapper shouldFetchOnAccountUpdate={modalOpen}>
```
Comment (just-toby) [inline src/components/CurrencyInputPanel/index.tsx:234]: just wrapping with this

File: src/components/CurrencyInputPanel/index.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 import { Pair } from '@uniswap/v2-sdk'
 import { useWeb3React } from '@web3-react/core'
 import { TraceEvent } from 'analytics'
+import PrefetchBalancesWrapper from 'components/AccountDrawer/PrefetchBalancesWrapper'
```
Comment (JFrankfurt) [inline src/components/CurrencyInputPanel/index.tsx:7]: path name might lead people to believe this is exclusively an account component--should prob bump it up to a root level component now that it's used here

**Review by JFrankfurt — CHANGES_REQUESTED** [review]
lgtm--1 request

**Review by natew — APPROVED** [review]
agree w @JFrankfurt otherwise lgtm

**Review by JFrankfurt — APPROVED** [review]

**Pull request merged**